### PR TITLE
Add Planetscale chat message history

### PR DESCRIPTION
**Pull request by milanjrodd**
Adds Planetscale chat message history by utilizing `@planetscale/database` pkg

To run tests:
- add Planetscale credentials to .env file
- run `yarn test:single -- planetscale.int.test.ts`

All tests were passed successfully.

Similar to dynamodb
https://github.com/hwchase17/langchainjs/pull/1016

### CODE REVIEW

File: docs/extras/modules/memory/integrations/planetscale.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+hide_table_of_contents: true
+---
+
+import CodeBlock from "@theme/CodeBlock";
+
+# Planetscale Chat Memory
```
Comment (jacoblee93) [inline docs/extras/modules/memory/integrations/planetscale.mdx:7]: Looks like they prefer `PlanetScale` (camel-cased):
https://planetscale.com/

File: langchain/package.json [hunk 2]
```diff
@@ -610,6 +613,7 @@
     "@notionhq/client": "^2.2.5",
     "@opensearch-project/opensearch": "*",
     "@pinecone-database/pinecone": "*",
+    "@planetscale/database": "^1.8.0",
```
Comment (jacoblee93) [inline langchain/package.json:616]: Can you add as a `devDependency` as well?

File: langchain/src/stores/message/planetscale.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      this.client = new PlanetscaleClient(config);
+    } else {
+      throw new Error(
+        "Either a client or config must be provided to PlanetscaleChatMessageHistory"
+      );
+    }
+
+    this.connection = this.client.connection();
+
+    this.tableName = tableName || "stored_message";
```
Comment (jacoblee93) [inline langchain/src/stores/message/planetscale.ts:71]: nit: How about `langchain_chat_histories` for the default?

File: langchain/src/stores/message/planetscale.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    this.sessionId = sessionId;
+  }
+
+  private async createTableIfNotExists(): Promise<void> {
+    const query = `CREATE TABLE IF NOT EXISTS ${this.tableName} (id BINARY(16) PRIMARY KEY, session_id VARCHAR(255), type VARCHAR(255), content VARCHAR(255), role VARCHAR(255), name VARCHAR(255), additional_kwargs VARCHAR(255));`;
```
Comment (jacoblee93) [inline langchain/src/stores/message/planetscale.ts:76]: nit: Store `additional_kwargs` as a JSON field for potential easier querying down the road?

**Review by jacoblee93 — COMMENTED** [review]
Looks great, a few nits, and we should probably call this `PlanetScale` to match their preferred name.

**Comment by jacoblee93** [comment]
Simple changes so went ahead and made them - thank you!

**Comment by milanjrodd** [comment]
Oh, that was fast 😀

I think there might be a problem with additional_learns field as a varchar255. So needs to be json type. 

Also there is no difference if table id would be type of int, so we can make one sql request instead of two, without altering the table.

Should I open an issue for that?

**Comment by jacoblee93** [comment]
Sure, though I ran the integration test and it seemed to work just fine

**Comment by milanjrodd** [comment]
I'm not sure if json of additional_kwargs always fits in varchar255. And I suppose there is no test for that special case. I mean it might be longer then 255 symbols.

I think sql optimization would be useful for serverless environments, where it can not hold tableCreated state between multiple functions.

That's my first contribution ever, should I create 2 different issues for that?

**Comment by jacoblee93** [comment]
Feel free to just submit another PR as well#

**Pull request merged**